.github/workflows/build-and-deploy.yml

```
@@ -50,3 +50,32 @@ jobs:
+    if: github.event_name == 'push' && github.ref == 'refs/heads/main' # Only run this job on push events to the main branch
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v4 # Check out the repository code
+
+      - name: Set up Git
+        run: |
+          git config --global user.name 'github-actions'
+          git config --global user.email 'github-actions@github.com'
+
+      - name: Update project version
+        run: mvn versions:set -DnewVersion=$(date +'%Y%m%d%H%M%S') # Update the project version using the current date and time
```

actionlint

shellcheck reported issue in this script: SC2046:warning:1:31: Quote this to prevent word splitting

(shellcheck)